core/src/HIP/Kokkos_HIP_Space.cpp

```diff
@@ -642,6 +642,9 @@ void SharedAllocationRecord<Kokkos::Experimental::HIPSpace, void>::
 /*--------------------------------------------------------------------------*/
 /*--------------------------------------------------------------------------*/
 namespace Kokkos {
+namespace Impl {
+extern int get_gpu(const InitArguments& args);
```

Comment:
Why not just forward-declare normally?
```suggestion
int get_gpu(const InitArguments& args);
```

Reply:
Agreed. I made this comment earlier. It's technically not wrong—`extern` is implicit here—but it's noise.